### PR TITLE
Make LCompilers CLI check if source file exists

### PR DESCRIPTION
**Pull request by faze-geek**
Closes https://github.com/lcompilers/lpython/issues/1490

On master
```
(lp) C:\Users\kunni\lpython>python wrong_name.py
python: can't open file 'C:\\Users\\kunni\\lpython\\wrong_name.py': [Errno 2] No such file or directory

(lp) C:\Users\kunni\lpython>src\bin\lpython wrong_name.py

(lp) C:\Users\kunni\lpython>src\bin\lpython --show-ast wrong_name.py
(Module [] [])
```

On branch-
```
(lp) C:\Users\kunni\lpython>src\bin\lpython wrong_name.py
Internal Compiler Error: Unhandled exception
Traceback (most recent call last):
LCompilersException: No such file or directory: wrong_name.py

(lp) C:\Users\kunni\lpython>src\bin\lpython --show-ast wrong_name.py
Internal Compiler Error: Unhandled exception
Traceback (most recent call last):
LCompilersException: No such file or directory: wrong_name.py

```

### CODE REVIEW

**Review by czgdp1807 — APPROVED** [review]
Much needed. Would ask @certik as well before merging.

**Comment by faze-geek** [comment]
cc @Thirumalai-Shaktivel

**Comment by Thirumalai-Shaktivel** [comment]
As this PR has the same changes as the one in lfortran: https://github.com/lfortran/lfortran/pull/741/files. I think this might be good to go in.

**Review by Thirumalai-Shaktivel — APPROVED** [review]

**Comment by czgdp1807** [comment]
In any case, please don't merge this before libasr sync is complete.

**Comment by Shaikh-Ubaid** [comment]
I think this is a good change.

**Comment by czgdp1807** [comment]
Seems to work for me,

```zsh
(lp) 18:11:40:~/lpython_project/lpython % lpython --show-asr foo.py
Internal Compiler Error: Unhandled exception
Traceback (most recent call last):
  File "/Users/czgdp1807/lpython_project/lpython/src/bin/lpython.cpp", line 1563
    throw LCompilers::LCompilersException("No such file or directory: " + arg_file);
LCompilersException: No such file or directory: foo.py
(lp) 18:11:49:~/lpython_project/lpython % lpython --show-llvm foo.py
Internal Compiler Error: Unhandled exception
Traceback (most recent call last):
  File "/Users/czgdp1807/lpython_project/lpython/src/bin/lpython.cpp", line 1563
    throw LCompilers::LCompilersException("No such file or directory: " + arg_file);
LCompilersException: No such file or directory: foo.py
(lp) 18:11:55:~/lpython_project/lpython % lpython foo.py            
Internal Compiler Error: Unhandled exception
Traceback (most recent call last):
  File "/Users/czgdp1807/lpython_project/lpython/src/bin/lpython.cpp", line 1563
    throw LCompilers::LCompilersException("No such file or directory: " + arg_file);
LCompilersException: No such file or directory: foo.py
```

@certik I will merge this tomorrow if you don't have any objections.

**Pull request merged**
